### PR TITLE
Add blog post for EKS Auto Mode

### PR DESCRIPTION
**Pull request by flostadler**
### Proposed changes

> Today at re:Invent, AWS announced Amazon Elastic Kubernetes Service (Amazon EKS) Auto Mode, a new feature that fully automates compute, storage, and networking management for Kubernetes clusters.

[AWS Announcement](https://aws.amazon.com/about-aws/whats-new/2024/12/amazon-eks-auto-mode/)
[AWS Blog post](https://aws.amazon.com/blogs/aws/streamline-kubernetes-cluster-management-with-new-amazon-eks-auto-mode/)

This adds a blog post about the new EKS Auto Mode feature AWS released and how users can leverage it with Pulumi.
This is the PR for adding support to pulumi-eks: https://github.com/pulumi/pulumi-eks/pull/1519

### TODO
- [x] Release EKS Auto Mode for pulumi-eks
- [x] Update `meta.png`

### CODE REVIEW

File: content/blog/aws-eks-auto-mode/index.md [hunk 1]
```diff
@@ -0,0 +1,2051 @@
...
+        enabled: true
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+## Deploying a Load-Balanced Application
+
+Let's deploy a simple web application fronted by an Application Load Balancer now that EKS Auto Mode is enabled.
```
Comment (guineveresaenger) [inline content/blog/aws-eks-auto-mode/index.md:389]:
```suggestion
Now that we have provisioned an EKS cluster with Auto Mode enabled, let's deploy a web application fronted by an Application Load Balancer.
```

File: content/blog/aws-eks-auto-mode/index.md [hunk 2]
```diff
@@ -0,0 +1,2051 @@
...
+EKS Auto Mode will automatically provisions appropriately sized nodes for your workload, set up the Application Load Balancer according
```
Comment (guineveresaenger) [inline content/blog/aws-eks-auto-mode/index.md:390]:
```suggestion
EKS Auto Mode will automatically provision appropriately sized nodes for your workload, set up the Application Load Balancer according
```

File: content/blog/aws-eks-auto-mode/index.md [hunk 3]
```diff
@@ -0,0 +1,2051 @@
...
+to the Ingress configuration and manage ongoing maintenance and updates.
```
Comment (guineveresaenger) [inline content/blog/aws-eks-auto-mode/index.md:391]:
```suggestion
to the Ingress configuration, and manage ongoing maintenance and updates.
```

(sorry I think the Oxford comma makes for easier reading here! 🙈 )

File: content/blog/aws-eks-auto-mode/index.md [hunk 4]
```diff
@@ -0,0 +1,2051 @@
...
+After deploying this with `pulumi up`, you can access your application with `curl` once the Load Balancer has been provisioned:
+
+```bash
+~ curl $(pulumi stack output url)
+Hello, Pulumi!
+```
+
+## Deploying a Stateful Workload
+
+EKS Auto Mode simplifies persistent storage management by automatically handling EBS volume provisioning for your stateful workloads. To leverage this feature you can create a `StorageClass` that uses the EBS CSI driver (`ebs.csi.eks.amazonaws.com`) as its provisioner. EKS Auto Mode will then automatically manage the lifecycle of your persistent volumes, including provisioning, attaching, and cleanup.
```
Comment (guineveresaenger) [inline content/blog/aws-eks-auto-mode/index.md:1343]:
```suggestion
EKS Auto Mode simplifies persistent storage management by automatically handling EBS volume provisioning for your stateful workloads. To leverage this feature, you can create a `StorageClass` that uses the EBS CSI driver (`ebs.csi.eks.amazonaws.com`) as its provisioner. EKS Auto Mode will then automatically manage the lifecycle of your persistent volumes, including provisioning, attaching, and cleanup.
```

File: content/blog/aws-eks-auto-mode/index.md [hunk 5]
```diff
@@ -0,0 +1,2051 @@
...
+EKS Auto Mode represents a significant step forward in simplifying Kubernetes operations on AWS. Pulumi's support for this feature allows you
+to define EKS Auto Mode clusters using your favorite programming language and integrate them with your existing infrastructure code.
+
+To get started with EKS Auto Mode in Pulumi:
+
+1. Update your AWS providers to the latest versions
+2. Check out our [EKS documentation](/registry/packages/eks/)
+3. Join our [Community Slack](https://slack.pulumi.com/) for support and discussions
+
+We are excited to see what you build with EKS Auto Mode and Pulumi!
```
Comment (guineveresaenger) [inline content/blog/aws-eks-auto-mode/index.md:2051]: love this entire summary! 💟

File: content/blog/aws-eks-auto-mode/index.md [hunk 6]
```diff
@@ -0,0 +1,2051 @@
...
+    - kubernetes
+---
+
+AWS recently announced [Amazon EKS Auto Mode](https://aws.amazon.com/eks/auto-mode/), a significant enhancement to Amazon EKS that streamlines cluster management by automating infrastructure decisions and operations. Today, we are excited to announce that Pulumi fully supports EKS Auto Mode across our AWS provider ecosystem, enabling you to leverage this powerful feature through infrastructure as code.
+
+<!--more-->
+
+## What is EKS Auto Mode?
+
+EKS Auto Mode extends AWS management beyond the Kubernetes control plane to the underlying infrastructure that powers your workloads. It automates critical aspects of cluster operations including:
```
Comment (guineveresaenger) [inline content/blog/aws-eks-auto-mode/index.md:21]: I think that despite linking out to the AWS docs, this section could handle a bit more detail on how it automates k8s clusters, and how that makes things easier and faster.

The Pulumi wrapper is really cool, definitely don't mean to knock it - but from the Pulumi side all the magic lies in the fact that yes, you can use Pulumi for this. Hence, it doesn't hurt to do a secondary plug for "why would I want to use Auto Mode anyway".

File: content/blog/aws-eks-auto-mode/index.md [hunk 7]
```diff
@@ -0,0 +1,2051 @@
...
+
+This automation significantly reduces the operational overhead of running Kubernetes clusters, allowing teams to focus on application development rather than infrastructure management.
+
+## Pulumi Support for EKS Auto Mode
+
+We have added EKS Auto Mode support across our AWS provider ecosystem:
+
+- AWS Native Provider (v6.63.0+)
+- AWS Cloud Control Provider (v1.13.0+)
+- Pulumi EKS Provider (TODO: upcoming release)
```
Comment (guineveresaenger) [inline content/blog/aws-eks-auto-mode/index.md:37]: Should we mention AWS Classic here?

**Review by guineveresaenger — COMMENTED** [review]
nice first draft! extreme grammar nits and some thoughts; none blocking.

File: content/blog/aws-eks-auto-mode/index.md (hunk 7 above)
Comment (flostadler) [reply content/blog/aws-eks-auto-mode/index.md:37]: Uh good catch, the first item is actually classic and I called it native 🤦

File: content/blog/aws-eks-auto-mode/index.md (hunk 6 above)
Comment (flostadler) [reply content/blog/aws-eks-auto-mode/index.md]: Good point! I added some more details to this section

File: content/blog/aws-eks-auto-mode/index.md [hunk 8]
```diff
@@ -0,0 +1,2059 @@
...
+        {type: SubnetType.Private, tags: {[`kubernetes.io/cluster/${clusterName}`]: "shared", "kubernetes.io/role/internal-elb": "1"}},
+    ],
+    subnetStrategy: "Auto"
+});
+
+const cluster = new eks.Cluster("eks-auto-mode", {
+    name: clusterName,
+    // EKS Auto Mode requires Access Entries, use either the `Api` or `ApiAndConfigMap` authentication mode.
+    authenticationMode: eks.AuthenticationMode.Api,
+    skipDefaultNodeGroup: true,
```
Comment (corymhall) [inline content/blog/aws-eks-auto-mode/index.md:88]: Should we remove these 2 defaults from the example?

**Review by corymhall — APPROVED** [review]

File: content/blog/aws-eks-auto-mode/index.md (hunk 8 above)
Comment (flostadler) [reply content/blog/aws-eks-auto-mode/index.md:88]: Good point, missed updating the examples after updating the PR!

File: content/blog/aws-eks-auto-mode/index.md [hunk 9]
```diff
@@ -0,0 +1,2047 @@
...
+- Managing storage drivers and volume provisioning
+- Handling node updates and security patches
+- Optimizing instance selection and placement
+
+## Pulumi Support for EKS Auto Mode
+
+We have added EKS Auto Mode support across our AWS provider ecosystem:
+
+- Pulumi AWS Provider (v6.63.0+)
+- Pulumi AWS Cloud Control Provider (v1.13.0+)
```
Comment (meagancojocar) [inline content/blog/aws-eks-auto-mode/index.md:44]:
```suggestion
- [Pulumi AWS Cloud Control Provider](/registry/packages/aws-native/) (previously known as Pulumi AWS Native) (v1.13.0+)
```

File: content/blog/aws-eks-auto-mode/index.md [hunk 10]
```diff
@@ -0,0 +1,2047 @@
...
+- Configuring and updating networking plugins
+- Managing storage drivers and volume provisioning
+- Handling node updates and security patches
+- Optimizing instance selection and placement
+
+## Pulumi Support for EKS Auto Mode
+
+We have added EKS Auto Mode support across our AWS provider ecosystem:
+
+- Pulumi AWS Provider (v6.63.0+)
```
Comment (meagancojocar) [inline content/blog/aws-eks-auto-mode/index.md:43]:
```suggestion
- [Pulumi AWS Provider](/registry/packages/aws/) (v6.63.0+)
```

File: content/blog/aws-eks-auto-mode/index.md [hunk 11]
```diff
@@ -0,0 +1,2047 @@
...
+- Pulumi AWS Cloud Control Provider (v1.13.0+)
+- Pulumi EKS Provider (TODO: upcoming release)
```
Comment (meagancojocar) [inline content/blog/aws-eks-auto-mode/index.md:45]:
```suggestion
- [Pulumi EKS Provider ](/registry/packages/eks/)(TODO: upcoming release)
```

**Review by meagancojocar — APPROVED** [review]
This is so great! Good work team

**Review by t0yv0 — APPROVED** [review]

**Pull request merged**